src/Campaigns/Blocks/shared/hooks/useCampaigns.ts

```diff
@@ -1,11 +1,38 @@
 import {useEntityRecords} from '@wordpress/core-data';
 import {Campaign} from '@givewp/campaigns/admin/components/types';
 
-export default function useCampaigns() {
-    const data = useEntityRecords('givewp', 'campaign');
+type useCampaignsParams = {
+    ids?: number[],
+    page?: number,
+    per_page?: number;
+    status?: 'active' | 'draft' | 'archived';
+    sortBy?: string;
+    orderBy?: string;
+}
+
+export default function useCampaigns({
+     ids = [],
+     page = 1,
+     per_page = 30,
+     status = 'active',
+     sortBy = 'date',
+     orderBy = 'desc',
+ }: useCampaignsParams = {}) {
+    const data = useEntityRecords('givewp', 'campaign', {
+        ids,
+        page,
+        per_page,
+        status,
+        sortBy,
+        orderBy
+    });
 
     return {
         campaigns: data?.records as Campaign[],
+        //@ts-ignore
```

Comment:
Would be nice to figure out how to resolve this typescript issue

Reply:
There is no interface or type I can import, at least I don't know about it. I spent quite some time on this. This is a quick fix for now.

Reply:
I hear you, these WP hooks can be a challenge.  I'll do some digging